### PR TITLE
Use an enum for PNC operation codes

### PR DESCRIPTION
**Pull request by BushraAbdullahi**
named the enums based on names used in [original Java code](https://github.com/ministryofjustice/bichard7-next/blob/main/bichard-pnc/src/main/java/uk/gov/ocjr/mtu/br7/pnc/types/PNCMessageType.java)

### CODE REVIEW

**Review by chubberlisk — COMMENTED** [review]
Left some comments - I'd recommend you rebase (instead of merging main) your branch as it has a lot of unrelated commits which makes the PR noisy.

File: packages/core/types/PncOperationCodes.ts
[the remark this answers is not in the document]
Comment (BushraAbdullahi) [reply packages/core/types/PncOperationCodes.ts:1]: these were from Sen when he first picked up the ticket. I forgot to remove them

File: packages/core/phase2/lib/getOperationSequence/validateOperations.ts [hunk 1]
```diff
@@ -44,41 +45,52 @@ const validateOperations = (operations: Operation[], remandCcrs: Set<string>): E
 
     if (penhrgExists && courtCaseSpecificOperations.length > 0) {
       const incompatibleCode = courtCaseSpecificOperations[courtCaseSpecificOperations.length - 1].code
-      if (["SUBVAR", "PENHRG"].includes(incompatibleCode)) {
+      if ([PncOperation.DISPOSAL_UPDATED, PncOperation.PENALTY_HEARING].includes(incompatibleCode as PncOperation)) {
```
Comment (chubberlisk) [inline packages/core/phase2/lib/getOperationSequence/validateOperations.ts:48]: Do we need the casting?

File: packages/core/phase2/lib/getOperationSequence/validateOperations.ts [hunk 2]
```diff
@@ -44,41 +45,52 @@ const validateOperations = (operations: Operation[], remandCcrs: Set<string>): E
...
 
-    if (["APPHRD", "COMSEN", "SENDEF", "SUBVAR", "DISARR"].includes(operation.code)) {
+    if (
+      [
+        PncOperation.APPEALS_UPDATE,
+        PncOperation.COMMITTED_SENTENCING,
+        PncOperation.SENTENCE_DEFERRED,
+        PncOperation.DISPOSAL_UPDATED,
+        PncOperation.NORMAL_DISPOSAL
+      ].includes(operation.code as PncOperation)
```
Comment (chubberlisk) [inline packages/core/phase2/lib/getOperationSequence/validateOperations.ts:66]: Do we need the casting?

File: packages/core/phase2/lib/getOperationSequence/validateOperations.ts [hunk 3]
```diff
@@ -90,11 +102,14 @@ const validateOperations = (operations: Operation[], remandCcrs: Set<string>): E
...
     const remandCcrsContainCourtCaseReference = !!courtCaseReference && remandCcrs.has(courtCaseReference)
 
-    if (operation.code === "APPHRD" && remandCcrsContainCourtCaseReference) {
+    if (operation.code === PncOperation.APPEALS_UPDATE && remandCcrsContainCourtCaseReference) {
       return { code: ExceptionCode.HO200109, path: errorPath }
     }
 
-    if (["COMSEN", "SENDEF"].includes(operation.code) && remandCcrsContainCourtCaseReference) {
+    if (
+      [PncOperation.COMMITTED_SENTENCING, PncOperation.SENTENCE_DEFERRED].includes(operation.code as PncOperation) &&
```
Comment (chubberlisk) [inline packages/core/phase2/lib/getOperationSequence/validateOperations.ts:110]: Do we need the casting?

File: packages/core/phase2/lib/getOperationSequence/validateOperations.ts (hunk 2 above)
Comment (BushraAbdullahi) [reply packages/core/phase2/lib/getOperationSequence/validateOperations.ts:66]: it was showing errors without it

File: packages/core/phase2/lib/getOperationSequence/validateOperations.ts (hunk 3 above)
Comment (BushraAbdullahi) [reply packages/core/phase2/lib/getOperationSequence/validateOperations.ts:110]: it was showing errors without it

**Pull request closed without merging**